### PR TITLE
update dashboards and other operational changes

### PR DESCRIPTION
**Pull request by triclambert**
* Fix dashboards
  - [x] Update 'container_name' label
  - [x] Make tenant list generated dynamically
  - [x] Add PVC graphs to overview dashboard
  - [x] Add pod detail graphs
  - [x] Remove swap graph and pod states graph
  - [x] Polish things here and there (e.g., remove legends where messy)
  - [x] Strip `-tenant` from the label_values results on some dashboards, as it was breaking them
  - [x] Remove unused dashboards

### CODE REVIEW

File: packages/cli/src/util.ts [hunk 1]
```diff
@@ -32,8 +32,8 @@ import * as list from "./list";
  * messages.
  */
 export function validateClusterNameOrDie(cn: string): void {
-  if (cn.length > 13) {
-    die(`cluster name must not be longer than 13 characters`);
+  if (cn.length > 21) {
+    die(`cluster name must not be longer than 21 characters`);
```
Comment (triclambert) [inline packages/cli/src/util.ts:36]: @spahl @sreis @jgehrcke any issues with this?

Comment (MatApple) [reply packages/cli/src/util.ts:36]: It was originally set to 13 Chars because that was the limit for the BigTable instance name (when we had it for Loki/Cortex). Since that's no longer used then this should be fine. We may hit some issue elsewhere with another cloud service that has aggressive char limits on their Ids, since we use the cluster name in _most_ cloud infra Ids.

I'd suggest testing this with a cluster name that uses 21 chars in AWS and GCP

Comment (jgehrcke) [reply packages/cli/src/util.ts:36]: Let's please update https://github.com/opstrace/opstrace/issues/24

**Comment by triclambert** [comment]
Dashboards looking good.  Local build of the installer did not pick up the validation change; need to dig further.

**Comment by MatApple** [comment]
It’s the controller that bundles the dashboards

> 
> 
>

**Comment by triclambert** [comment]
Yup, the controller worked just fine.

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json [hunk 2]
```diff
@@ -158,7 +158,7 @@
       "reverseYBuckets": false,
       "targets": [
         {
-          "expr": "sum by (le) (rate(cortex_api_proxy_request_duration_seconds_bucket{job=\"$tenant-tenant.cortex-api\",method=\"GET\",status_code=\"200\",route=~\"api_prom_api_v1_query\"}[$timewindow]))",
+          "expr": "sum by (le) (rate(cortex_api_proxy_request_duration_seconds_bucket{job=\"$tenant-tenant.cortex-api\",method=\"POST\",status_code=\"200\",route=~\"api_v1_push\"}[$timewindow]))",
```
Comment (jgehrcke) [inline packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:161]: Why this change?

Comment (jgehrcke) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:161]: The center column in these dashboards is supposed to show the read direction, as indicated in the title of this panel:

```
API proxy /api/prom/api/v1/query, 200-acked GETs, request duration, 
```
changing this to keep track of POST requests would result in a pretty dangerous mismatch between title and actual plot :)

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
       "steppedLine": true,
       "targets": [
         {
-          "expr": "sum by (status_code) (rate(cortex_api_proxy_request_duration_seconds_count{job=\"$tenant-tenant.cortex-api\",method=\"POST\",route=\"api_prom_push\"}[$timewindow]))",
+          "expr": "sum by (status_code) (rate(cortex_api_proxy_request_duration_seconds_count{job=\"$tenant-tenant.cortex-api\",method=\"POST\",route=\"api_v1_push\"}[$timewindow]))",
```
Comment (jgehrcke) [inline packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:74]: can you update the title of this panel, too, please?

Comment (jgehrcke) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:74]: (it says `API proxy /api/prom/push` still right now)

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json [hunk 4]
```diff
@@ -2453,6 +2423,54 @@
...
+            "selected": false,
+            "text": "v3",
+            "value": "v3"
+          },
+          {
+            "selected": false,
+            "text": "launchpad",
+            "value": "launchpad"
+          }
+        ],
```
Comment (jgehrcke) [inline packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:2463]: I hope we can make this dynamic at some point! A drop-down, showing the actual tenants in the current cluster.

File: packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json [hunk 5]
```diff
@@ -348,7 +205,7 @@
       "timeFrom": null,
       "timeRegions": [],
       "timeShift": null,
-      "title": "[LOGS] API proxy /loki/api/v1/push, POSTs, rate by status code (5m mean)",
+      "title": "[LOGS] API proxy /loki/api/v1/push, POSTs, rate by status code (5m mean) ALL tenants",
```
Comment (jgehrcke) [inline packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json:208]: :rocket:

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json (hunk 2 above)
Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:161]: This was a mistake, reverting.

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json (hunk 4 above)
Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:2463]: This is a WIP.  The regex below does make this dynamic, but it saves the results here in the config.  I need to test omitting that part, or at least reverting back to "default".

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json [hunk 6]
```diff
@@ -643,7 +643,7 @@
       "steppedLine": true,
       "targets": [
         {
-          "expr": "sum by (status_code) (rate(cortex_api_proxy_request_duration_seconds_count{job=\"$tenant-tenant.cortex-api\",method=\"GET\",route=~\"api_prom_api_v1_query.*\"}[$timewindow]))",
+          "expr": "sum by (status_code) (rate(cortex_api_proxy_request_duration_seconds_count{job=\"$tenant-tenant.cortex-api\",method=\"GET\",route=~\"metrics\"}[$timewindow]))",
```
Comment (triclambert) [inline packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:646]: The only route available on the read path appears to be 'metrics'.  Is this expected @sreis @jgehrcke?

File: packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json [hunk 7]
```diff
@@ -545,26 +402,27 @@
         {
           "expr": "sum(rate(loki_distributor_lines_received_total[5m]))",
           "interval": "",
-          "legendFormat": "log entry rate",
+          "legendFormat": "entry rate",
           "refId": "A"
         },
         {
-          "expr": "avg(sum(rate(loki_distributor_lines_received_total[12h])))",
+          "expr": "sum(rate(loki_distributor_bytes_received_total[5m]))/sum(rate(loki_distributor_lines_received_total[5m]))",
```
Comment (triclambert) [inline packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json:409]: Is this "bytes per entry" calculation correct and useful?

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json (hunk 4 above)
Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:2453]: I think this is working now.  Will continue testing, and then update other dashboards.  Resolving this comment.

File: packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json (hunk 6 above)
Comment (jgehrcke) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:646]: The Grafana/browser based query path is expected to be a blind spot when using the data API proxy-based metrics! (those queries do not go through the data API proxies).

Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/cortex-overview-per-tenant.json:646]: Not quite following, are you saying that the `metrics` route is for browser-based queries?

Stepping back to the problem:  this graph appears empty without a change.  The only route that appears to be available is "metrics".

**Comment by triclambert** [comment]
@sreis I added a few panels to the overview dashboard based on our work today.  PTAL.

File: packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json (hunk 5 above)
Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json:208]: FWIW, I have now added a panel just like this that is "`BY tenants`", i.e., it is broken down by tenant rather than summed across them.  Please let me know how useable this is, especially the title of the panel.

File: packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json (hunk 7 above)
Comment (triclambert) [reply packages/controller/src/resources/monitoring/system/dashboards/opstrace-overview.json:628]: After using it a bit, we should keep it around but revisit in the future.
